### ISSUE TITLE
About layers and feature extraction on pre-trained models in new version

### ISSUE THREAD

**Issue by anselmoferreira** [vlfeat/matconvnet#828]
I used to apply an MNIST architecture-based pre-trained model (trained from the scratch with my own data) to extract feature vectors by extracting the features in the last but two layers. I used a very old matconvnet version (which I dont remember what is) and the following command (part of a for loop) extracted the features I want

```
res_train=vl_simplenn(net,batch_train_images)';
features_train=vertcat(features_train,squeeze(res_train(7).x)');
```
This means that I pick the outputs of layer 7 and use them as the feature vectors to be saved later.

But now, to do the same with the new matconvnet version I have to do the following:

```
res_train=vl_simplenn(net,batch_train_images)';
features_train=vertcat(features_train,squeeze(res_train(10).x)');
```
  Now I have to extract the output of layer 10.

What changed in the new version if cnn_mnist_init.m file remains the same?

**Comment by albanie** [comment]
It's hard to say exactly without knowing which version - `cnn_mnist_init.m` has had a bunch of [updates](https://github.com/vlfeat/matconvnet/commits/master/examples/mnist/cnn_mnist_init.m) through recent versions.  My guess would be that the extra layers are for batch normalization, but probably the best thing to do is to examine each layer separately and identify what has been changed.
